includes/paragraphs.inc

```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
+    if (!$item->entity->get('field_oe_image')->isEmpty()) {
+      /** @var \Drupal\file\Plugin\Field\FieldType\FileItem $image_item */
+      $image_item = $item->entity->get('field_oe_image')->first();
+      $file = $image_item->get('entity')->getValue();
+
+      $image = [
+        'path' => file_url_transform_relative(file_create_url($file->getFileUri())),
+        'alt' => $image_item->get('alt')->getValue(),
+      ];
+
+      set_image($variant, $image);
+    }
+    $title_content = $item->entity->field_oe_title->value;
+    $url_title = $item->entity->get('field_oe_link')->first()->getUrl()->toString();
+    if (!empty($url_title)) {
+      $title_content = [
+        '#markup' => '<a href="' . $url_title . '">' . $title_content . '</a>',
+      ];
+    }
+
+    // Prepare the date fields if date is available.
+    $contentItem = "";
+    if (!$item->entity->get('field_oe_date')->isEmpty()) {
+      $dateItem = $item->entity->get('field_oe_date')->first();
+      $contentItem = [
+        '#markup' => '<div><span class="d-md-inline d-block text-muted mb-2 mb-md-0">' . $dateItem->date->format("j F Y") . '</span></div>',
+      ];
+    }
+
+    // Prepare the metas if available.
+    $metas = [];
+    if (!$item->entity->get('field_oe_meta')->isEmpty()) {
+      foreach ($item->entity->get('field_oe_meta') as $item_meta) {
+        $metas[] = [
+          'label' => $item_meta->value,
+          'background' => 'primary',
+        ];
+      }
+    }
+
+    $variables['items'][] = [
+      'title' => [
+        'content' => $title_content,
+      ],
+      'text' => [
+        'content' => $item->entity->field_oe_text_long->value,
+        'classes' => $text_classes,
+      ],
+      'image' => $image,
+      'badges' => $metas,
+      'content' => $contentItem,
+    ];
+  }
+
+  // Prepare the button variables if a link has been specified.
+  if ($paragraph->get('field_oe_link')->isEmpty()) {
+    return;
+  }
+
+  /** @var \Drupal\link\Plugin\Field\FieldType\LinkItem $link_item */
+  $link_item = $paragraph->get('field_oe_link')->first();
+  $variables['link'] = [
+    'path' => $link_item->getUrl()->toString(),
+    'label' => $link_item->get('title')->getValue(),
+    'icon' => [
+      'path' => $variables['bcl_icon_path'],
+      'name' => 'chevron-right',
+    ],
+  ];
+}
+
+/**
+ * Sets variant block.
+ */
+function set_variant_block(&$variant_block): void {
+  switch ($variant_block) {
+    case 'one_column':
+      $variant_block = '1-col';
+      break;
+
+    case 'two_columns':
+      $variant_block = '2-col';
+      break;
+
+    case 'three_columns':
+      $variant_block = '3-col';
+      break;
+  }
+}
+
+/**
+ * Sets image position and rounded.
+ */
+function set_image($variant, array &$image): void {
+  if ($variant == 'thumbnail_secondary') {
+    $image['position'] = 'bottom';
+  }
+  elseif ($variant == 'thumbnail_secondary') {
+    $image['position'] = 'top';
+  }
+  else {
+    $image['rounded'] = 1;
```

Comment:
top or bottom or rounded ? we need a default one, top, then if variant is secondary show bottom